controllers/control/cassandratask_controller.go

```diff
@@ -0,0 +1,559 @@
+}
+
+func cleanup() TaskConfiguration {
+	return TaskConfiguration{
+		AsyncFeature: httphelper.AsyncSSTableTasks,
+		AsyncFunc:    callCleanup,
+		SyncFunc:     callCleanupSync,
+	}
+}
+
+func callRebuild(nodeMgmtClient httphelper.NodeMgmtClient, pod *corev1.Pod, args map[string]string) (string, error) {
+	return nodeMgmtClient.CallDatacenterRebuild(pod, args["source_datacenter"])
+}
+
+func rebuild(args map[string]string) TaskConfiguration {
+	return TaskConfiguration{
+		AsyncFeature: httphelper.Rebuild,
+		AsyncFunc:    callRebuild,
+		Arguments:    args,
+	}
+}
+
+func (r *CassandraTaskReconciler) getDatacenterPods(ctx context.Context, dc *cassapi.CassandraDatacenter) ([]corev1.Pod, error) {
+	var pods corev1.PodList
+
+	if err := r.Client.List(ctx, &pods, client.InNamespace(dc.Namespace), client.MatchingLabels(dc.GetDatacenterLabels())); err != nil {
+		return nil, err
+	}
+
+	return pods.Items, nil
+}
+
+func (r *CassandraTaskReconciler) reconcileEveryPodTask(ctx context.Context, taskId string, dc *cassapi.CassandraDatacenter, taskConfig TaskConfiguration) (ctrl.Result, error) {
+	logger := log.FromContext(ctx)
+
+	// We sort to ensure we process the dcPods in the same order
+	dcPods, err := r.getDatacenterPods(ctx, dc)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	sort.Slice(dcPods, func(i, j int) bool {
+		rackI := dcPods[i].Labels[cassapi.RackLabel]
+		rackJ := dcPods[j].Labels[cassapi.RackLabel]
+
+		if rackI != rackJ {
+			return rackI < rackJ
+		}
+
+		return dcPods[i].Name < dcPods[j].Name
+	})
+
+	nodeMgmtClient, err := httphelper.NewMgmtClient(ctx, r.Client, dc)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	for idx, pod := range dcPods {
+		features, err := nodeMgmtClient.FeatureSet(&pod)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+
+		if pod.Annotations == nil {
+			pod.Annotations = make(map[string]string)
+		}
+
+		jobStatus, err := GetJobStatusFromPodAnnotations(taskId, pod.Annotations)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+
+		podPatch := client.MergeFrom(pod.DeepCopy())
+		if jobStatus.Id != "" {
+			if jobStatus.Status == podJobCompleted {
+				// Next pod, this is done
+				continue
+			}
+			if features.Supports(taskConfig.AsyncFeature) {
+				// Pod is currently processing something, or has finished processing.. update the status
+				details, err := nodeMgmtClient.JobDetails(&pod, jobStatus.Id)
+				if err != nil {
+					logger.Error(err, "Could not get JobDetails for pod", "Pod", pod)
+					return ctrl.Result{}, err
+				}
+
+				if details.Id == "" {
+					// This job was not found, pod most likely restarted. Let's retry..
+					delete(pod.Annotations, getJobAnnotationKey(taskId))
+					err = r.Client.Patch(ctx, &pod, podPatch)
+					if err != nil {
+						return ctrl.Result{}, err
+					}
+					return ctrl.Result{RequeueAfter: 1 * time.Second}, nil
+				} else if details.Status == podJobError {
+					// Log the error, move on
```

Comment:
When the job finishes, either successfully or fails, the job annotation never gets removed from the pod so you end up with an accumulation of job annotations, e.g., 

```yaml
apiVersion: v1
kind: Pod
metadata:
  annotations:
    control.k8ssandra.io/job-0f003348-33db-4272-bfcd-790ad27785cf: '{"id":"0fc51790-c522-4f74-9d87-a93a08d252af","status":"COMPLETED","handler":"management-api"}'
    control.k8ssandra.io/job-91f0112a-5cf6-4e65-b889-54d6676d2d8d: '{"id":"c38a54e5-5a57-4d2e-9067-7957370466cf","status":"COMPLETED","handler":"management-api"}'
    control.k8ssandra.io/job-b7946ffd-2456-4544-bab8-c6f314c6fcc4: '{"id":"8b04c4f2-632b-429d-b678-aeeefe5489a6","status":"COMPLETED","handler":"management-api"}'
  creationTimestamp: "2021-12-13T18:29:52Z"
  generateName: test-dc2-default-sts-
```
We should be removing those once it is safe to do so.

Since the annotation refers to the CassandraTask uid, do you think it might provide more clarity to change the prefix to `task`, e.g., `control.k8ssandra.io/task-b7946ffd-2456-4544-bab8-c6f314c6fcc4`?